bash_scripts/aprime_ocnice_diags.bash

```diff
@@ -25,8 +25,14 @@ unset tmp_currentdir tmp_gittopdir
 export config_file="$log_dir/config.ocnice.$uniqueID"
 python python/setup_ocnice_config.py
 if [ $? -ne 0 ]; then
+  echo
   echo "Failed to build config.ocnice"
   exit 1
 fi
 
 python python/MPAS-Analysis/run_mpas_analysis.py $config_file
+if [ $? -ne 0 ]; then
+  echo
+  echo "Failed ocean/ice diagnostics"
+  exit 1
+fi
```

Comment:
This did not help. One of my tests failed in the streamfunctionMOC and this did not pick it up. So, that means that, if one of the mpas-analysis tasks fail, it will not picked up by this exit status.

Reply:
@milenaveneziani, could you tell me more about the error that you saw where this was not triggered?  Was it in setup_and_check() of a task?  If setup_and_check() fails, it doesn't cause MPAS-Analysis to fail because we consider this to potentially be caused by a run that simply isn't configured for a given task, e.g. a melt-rate task in a run without ice-shelf cavities.

According to these lines, we should be exiting with error code 1 if any tasks failed during the run_task() stage:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/v0.6/run_mpas_analysis#L351-L355

Reply:
I don't think it was the setup_and_check stage, but I may be wrong. Here is the error:
```
Plotting streamfunction of Meridional Overturning Circulation (MOC)...
*** MOC Analysis Member is on ***
...but not yet supported. Using offline MOC

  Reading region and transect mask for Atlantic...

  Compute and/or plot post-processed MOC climatological streamfunction...
   Read previously computed MOC streamfunction from file...

  Compute and/or plot post-processed Atlantic MOC time series...
   Load data...
   Read in previously computed MOC time series
analysis task streamfunctionMOC failed during run 
Traceback (most recent call last):
  File "/global/u2/m/milena/a-prime/python/MPAS-Analysis/mpas_analysis/shared/analysis_task.py", line 323, in run
    self.run_task()
  File "/global/u2/m/milena/a-prime/python/MPAS-Analysis/mpas_analysis/ocean/streamfunction_moc.py", line 171, in run_task
    dsMOCTimeSeries = self._compute_moc_time_series_postprocess()
  File "/global/u2/m/milena/a-prime/python/MPAS-Analysis/mpas_analysis/ocean/streamfunction_moc.py", line 495, in _compute_moc_time_series_postprocess
    dsMOCIn.year[inIndex].values == years,
  File "/global/homes/m/milena/proj_acme/milena/miniconda2.7/lib/python2.7/site-packages/xarray/core/common.py", line 168, in __getattr__
    (type(self).__name__, name))
AttributeError: 'Dataset' object has no attribute 'year'
```
it was due to a previous abort of the analysis, so the MOCtimeseries file was created but it had incorrect information in it. The problem went away once I purged that file.

Reply:
@milenaveneziani, were you running in serial mode or in parallel?  If in parallel, what does the log file for the parallel job look like?  That's where we would potentially see one of the exit statements I linked above.

Reply:
no, it was in serial mode.

I can try to generate cases with problems and see how this behaves.

Reply:
I created a quick test:
```bash
#!/bin/bash
python run_mpas_analysis.py config.QU240 --generate=climatologyMapSST
echo $?
```
Then, I messed up `run_task` somewhere on purpose so it would crash.  I see the following output:
```
$ ./temp.bash 
/home/xylar/code/mpas-work/analysis/mpas_analysis_repo/mpas_analysis/shared/climatology/mpas_climatology_task.py:298: UserWarning: climatology start and/or end year different from requested
requestd: 0002-0050
actual:   0002-0005

  warnings.warn(message)
Running mpasClimatologyOcean
   Task mpasClimatologyOcean has finished successfully.
Running climatologyMapSST: remapMpasClimatology
   Task climatologyMapSST: remapMpasClimatology has finished successfully.
Running climatologyMapSST
ERROR in task climatologyMapSST.  See log file /media/xylar/extra_data/analysis/output/GMPAS-QU240/baseline/logs/climatologyMapSST.log for details
There were errors in task climatologyMapSST
1
```
This is exactly what I would expect to see and it seems like I'm getting exit code 1, as expected.  So far, I'm not sure why your test with a messed up MOC file did something different.

Reply:
@milenaveneziani, thanks.  In serial I can reproduce your problem:
```
$ ./temp.bash 
/home/xylar/code/mpas-work/analysis/mpas_analysis_repo/mpas_analysis/shared/climatology/mpas_climatology_task.py:298: UserWarning: climatology start and/or end year different from requested
requestd: 0002-0050
actual:   0002-0005

  warnings.warn(message)

Computing MPAS climatologies from files:
    mpaso.hist.am.timeSeriesStatsMonthly.0002-01-01.nc through
    mpaso.hist.am.timeSeriesStatsMonthly.0005-12-01.nc
Execution time: 0:00:00.02

Remapping climatology sst
Execution time: 0:00:00.06
analysis task climatologyMapSST failed during run 
Traceback (most recent call last):
  File "/home/xylar/code/mpas-work/analysis/mpas_analysis_repo/mpas_analysis/shared/analysis_task.py", line 326, in run
    self.run_task()
  File "/home/xylar/code/mpas-work/analysis/mpas_analysis_repo/mpas_analysis/ocean/climatology_map.py", line 226, in run_task
    crash
NameError: name 'crash' is not defined

Execution time: 0:00:00.00
Generating webpage for viewing results...
0
```
I don't immediately understand why the error that is being thrown doesn't cause an exit with non-zero code but presumably it's something to do with multiprocessing or me capturing the output or something like that.  I'll look into it tomorrow.

Reply:
Clearly something to fix on the MPAS-Analysis side and make a quick patch for.

Reply:
thanks for looking into this @xylar.

Reply:
This issue has (hopefully) been addressed in https://github.com/MPAS-Dev/MPAS-Analysis/pull/283

Reply:
Your changes here look good.  I think what you had before would have worked but this is cleaner and clearer.

Reply:
yes, I agree with both statements. Since I got confused with identifying the right exit status during testing, I thought of making it cleaner.